bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -28,25 +26,52 @@ spec:
         openshift.io/component: network
         kubernetes.io/os: "linux"
     spec:
-      # Requires fairly broad permissions - ability to read all services and network functions as well
-      # as all pods.
       serviceAccountName: ovn-kubernetes-controller
       hostNetwork: true
-      hostPID: true
       priorityClassName: "system-cluster-critical"
+      # volumes:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch - ovsdb schema (selinux blocks /etc on host)
+      # /var/lib/openvswitch -> /var/lib/openvswitch - ovsdb data
+      # /var/run/openvswitch -> /var/run/openvswitch - ovsdb sockets
+      #
       containers:
-      # firewall rules for ovn - assumed to be setup
-      # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6641 -j ACCEPT
-      # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6642 -j ACCEPT
-      # ovs flow for ovn (geneve)
-      # /usr/share/openvswitch/scripts/ovs-ctl --protocol=udp --dport=6081 enable-protocol
-
-      # run-ovn-northd - v3
-      - name: run-ovn-northd
+      - name: northd
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "run-ovn-northd"]
-
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          retries=0
+          nb_ok=0
+          sb_ok=0
+          while true; do
+            if ovn-nbctl show; then
+              nb_ok=1
+            else
+              echo "nbdb not yet up"
+              nb_ok=0
+            fi
+            if ovn-sbctl show; then
+              sb_ok=1
+            else
+              echo "sbdb not yet up"
+              sb_ok=0
+            fi
+            if (( nb_ok && sb_ok )); then
+              break
+            fi
+            sleep 5
+            (( retries += 1))
+            if [[ "${retries}" -gt 10 ]]; then
+              echo "timed out waiting for local databases"
+              exit 1
+            fi
+          done
+          ovn-nbctl set-connection ptcp:${OVN_NB_PORT} -- set connection . inactivity_probe=0
+          ovn-sbctl set-connection ptcp:${OVN_SB_PORT} -- set connection . inactivity_probe=0
+          exec ovn-northd --no-chdir -vconsole:info -vfile:off --pidfile=/var/run/openvswitch/ovn-northd.pid
```

Comment:
@squeed would be useful to have env vars for log levels for all these components so we can change that more easily when we need to.

Reply:
agreed - I'll add the sysconfig overrides for now, and we can think about a configmap that allows per-host overrides.